### PR TITLE
Add support for socks5

### PR DESCRIPTION
**Pull request by andyhu**
Please refer to the following issues:
TooTallNate/superagent-proxy#11
lapwinglabs/x-ray#1

Need to use together with TooTallNate/node-proxy-agent#5

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "socks-proxy-agent",
-  "version": "0.1.2",
-  "description": "A SOCKS (v4a) proxy `http.Agent` implementation for HTTP and HTTPS",
+  "version": "0.2.0",
```
Comment (TooTallNate) [inline package.json]: Please don't bump the version number. I'll do that in a separate commit.

Comment (andyhu) [inline package.json]: OK, I will fix it

**Comment by TooTallNate** [comment]
Very cool! I'm thinking that the diff would be smaller though if we instead utilized the `SocksClient.createConnection()` function like in the README. That way we bypass their Agent implementation completely, and maintain the `agent-base` guarantees ([it looks like their SocksAgent is loosely based off of `agent-base` anyways](https://github.com/JoshGlazebrook/socks-client/blob/1c7e92f2a8012f4e691d408c91e471fc4a984263/lib/socks-agent.js#L63), so might as well stick with the official one :smile:).

**Comment by andyhu** [comment]
@TooTallNate Yes I totally agree, if we can use agent-base for all 3 adepters it will be better. However I haven't got enough time working on it. It might need some copy and paste work instead of using it entirely. But there might be a chance to break something, I'm not quite familiar with socks protocol anyway.

**Comment by TooTallNate** [comment]
Take a look at 3525036. I was able to get things going pretty easily it seems like.

**Comment by TooTallNate** [comment]
Superseded by #4.

**Comment by TooTallNate** [comment]
Thanks for the nudge, btw :smile:

**Pull request closed without merging**